src/ExternalApiExamples/Program.cs

```diff
@@ -1,23 +1,52 @@
-﻿using System;
+﻿using Kmd.Logic.Identity.Authorization;
+using Microsoft.Extensions.Configuration;
+using System;
+using System.Net.Http;
 using System.Threading.Tasks;
-using ExternalApiExamples.Clients;
 
 namespace ExternalApiExamples
 {
     internal class Program
     {
         private static async Task Main(string[] args)
+        {
+            try
+            {
+                var config = new ConfigurationBuilder()
+                    .SetBasePath(AppContext.BaseDirectory)
+                    .AddJsonFile("appsettings.json", optional: false)
+                    .AddEnvironmentVariables()
+                    .Build()
+                    .Get<AppConfiguration>();
+
+                await Run(config).ConfigureAwait(false);
+            }
+#pragma warning disable CA1031 // Do not catch general exception types
+            catch (Exception ex)
+            {
+                Console.WriteLine(ex.Message);
+            }
+#pragma warning restore CA1031 // Do not catch general exception types
+        }
+
+        private static async Task Run(AppConfiguration configuration)
         {
             Console.WriteLine("KMD Studica Examples");
-            var tokenProvider = new StudicaHostToHostTokenProvider("-- SECRET --");
-            await new StudentsExample().Execute(tokenProvider);
-            await new EmployeeExample().Execute(tokenProvider);
-            await new RoomsExample().Execute(tokenProvider);
-            await new SchoolHoursPlansExample().Execute(tokenProvider);
-            await new SchoolHourEntryExample().Execute(tokenProvider);
-            await new StudentsExample().Execute(tokenProvider);
-            await new EducationalProgrammesExample().Execute(tokenProvider);
-            await new SubjectCoursesExample().Execute(tokenProvider);
+
+            using var httpClient = new HttpClient();
+            var tokenProvider = new LogicTokenProviderFactory(configuration.TokenProvider).GetProvider(httpClient);
+
+            using var externalApiHttpClient = new HttpClient();
+            externalApiHttpClient.BaseAddress = configuration.StudicaExternalApiBaseAddress;
+            externalApiHttpClient.DefaultRequestHeaders.Add("logic-api-key", configuration.StudicaExternalApiKey);
```

Comment:
Shouldn't matter but can we set it to "Logic-Api-Key" please?

Reply:
fixed